### DISCUSSION TITLE
[Idea]  SEO Urls Optimization

### DISCUSSION THREAD

**Discussion by Maofree** [PrestaShop/PrestaShop#37481] [Ideas]
### The problem

Hi
I also suggest to move at the end the IDs like so
{rewrite}{-:ean13}-{id}{-:id_product_attribute}.html

this way the numbers will not influence the initial part of the urls.
According to Google the initial part is always more important than the final part, this applies to titles, paragraphs etc...
many ask for the removal of the id from the urls, but in the end it is enough to move them to the end and make this thing the default I think it is a good thing

### The solution

for product
{rewrite}{-:ean13}-{id}{-:id_product_attribute}.html

the same for category
{rewrite}-{id}
for supplier
supplier/{rewrite}-{id}
and brand
brand/{rewrite}-{id}

### Do you plan to work on this subject?

- [ ] I'm willing to contribute a [formal specification](https://github.com/PrestaShop/prestashop-specs).
- [ ] I'm willing to provide any wireframes or design assets required for this feature.
- [ ] I'm willing to submit a Pull Request that implements this feature.
- [X] I'm willing to help verify that the implemented feature works as intended and produces no unintended side effects.

**Comment by jeckyl** [comment]
As URL structure has no influence on page referencing, the choice of ID location is purely aesthetic.
Removing the ID from the URL is no more important, and causes more technical problems than possible improvements.
I've just done an [article on the subject of modules for removing IDs](https://www.mediacom87.com/the-scam-of-modules-to-remove-ids-from-prestashop-urls/).

**Comment by Maofree** [comment]
in fact I suggested moving them to the end

**Comment by dheerajsukumar** [comment]
Solution

Product: {rewrite}{-:ean13}-{id}{-:id_product_attribute}.html
Category: {rewrite}-{id}
Supplier: supplier/{rewrite}-{id}
Brand: brand/{rewrite}-{id}

makes the URLs cleaner while keeping the IDs for functionality. Plus, it avoids the complexities of removing IDs altogether. Setting this as a default is definitely a practical solution!

Best,
Dheeraj
[SEO Expert in Kerala](https://dheerajsukumar.com/about-dheeraj/)

**Comment by darkf3d3** [comment]
I see no reason to move the id to the end.
For me, either remove the id completely or keep it as it is.
